### PR TITLE
#2038 Comment, service type, patient type

### LINKED ISSUE #2038: PaymentType, PatientType and ServiceType
## Is your feature request related to a problem? Please describe.

- Need to correctly save the payment type on the `RC` transaction for a prescription payment
- Need to choose the PatientType and ServiceType for a prescription 

## Describe the solution you'd like

N/A

## Implementation

N/A

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #2038 

## Change summary

- Adds `user1` to `Transaction` 🤷‍♂ 
- Adds an 'extra info' component housing the patient type, 
- Adds a few more module fields - most likely a `PreferencesReducer` will be needed soon. I wouldn't mind some sort of state which 'checks' for existence of records - i.e. when reasons are defined or programs then programs are enabled etc. having Preferences/modules and then I dunno what to call the last one - but it would be nice to be able to check a field, rather than query the length of some objects every time

image

- Top left thing here - needs work on styles will make a new issue this is big enough!

## Testing

N/A

### Related areas to think about

N/A

### CODE REVIEW

File: src/actions/PrescriptionActions.js [hunk 1]
```diff
@@ -14,8 +14,59 @@ import { WizardActions } from './WizardActions';
 export const PRESCRIPTION_ACTIONS = {
   REFRESH: 'Prescription/refresh',
   FILTER: 'Prescription/filter',
+  OPEN_COMMENT_MODAL: 'Prescription/openCommentModal',
+  CLOSE_COMMENT_MODAL: 'Prescription/closeCommentModal',
 };
 
+const editTransactionCategory = newValue => (dispatch, getState) => {
+  const { prescription } = getState();
+  const { transaction } = prescription;
```
Comment (wlthomson) [inline src/actions/PrescriptionActions.js]: Assuming this all required fields or safe getters? No defaults needed?

File: src/database/DataTypes/Transaction.js [hunk 2]
```diff
@@ -478,6 +478,7 @@ Transaction.schema = {
     insurancePolicy: { type: 'InsurancePolicy', optional: true },
     option: { type: 'Options', optional: true },
     linkedTransaction: { type: 'Transaction', optional: true },
+    user1: { type: 'string', optional: true },
```
Comment (wlthomson) [inline src/database/DataTypes/Transaction.js]: 😆 .

File: src/selectors/prescription.js [hunk 3]
```diff
@@ -74,3 +75,25 @@ export const selectSelectedRows = ({ prescription }) => {
 
   return items.reduce((acc, { item }) => ({ ...acc, [item.id]: true }), {});
 };
+
+export const selectTransactionCategoryName = ({ prescription }) => {
+  const { transaction } = prescription;
+  const { category } = transaction || {};
+  const { name } = category || {};
+  return name || '';
```
Comment (wlthomson) [inline src/selectors/prescription.js]: Nit: usual `??` nit.

File: src/widgets/PrescriptionExtra.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        />
+      )}
+      <PageInfo columns={pageInfoColumns} />
+      <ModalContainer
+        fullScreen
+        onClose={onCloseCommentModal}
+        isVisible={commentModalOpen}
+        title={modalStrings.edit_the_prescription_comment}
+      >
+        <TextEditor onEndEditing={x => onEditComment(x)} text={comment} />
```
Comment (wlthomson) [inline src/widgets/PrescriptionExtra.js]: Can this just be `onEndEditing={onEditComment}`?

File: src/actions/PrescriptionActions.js (hunk 1 above)
Comment (joshxg) [reply src/actions/PrescriptionActions.js]: ended up putting them in in a branched off pr hehe

File: src/database/DataTypes/Transaction.js (hunk 2 above)
Comment (joshxg) [reply src/database/DataTypes/Transaction.js]: 🤷‍♂

File: src/selectors/prescription.js (hunk 3 above)
Comment (joshxg) [reply src/selectors/prescription.js]: 5b7e41c294a80c037c7705d64b0f9ab0458eaa63

File: src/widgets/PrescriptionExtra.js (hunk 4 above)
Comment (joshxg) [reply src/widgets/PrescriptionExtra.js]: 1142a900f96f9b201f5068ec2d2589be2be09ff8 oops

**Pull request merged**